lib/recurly/pricing/index.js

```diff
@@ -243,14 +252,18 @@ Pricing.prototype.coupon = function (couponCode, done) {
 
   return new PricingPromise(function (resolve, reject) {
     if (!self.items.plan) return reject(errors('missing-plan'));
-    if (coupon) {
-      if (coupon.code === couponCode) return resolve(coupon);
-      else self.remove({ coupon: coupon.code });
+    if (coupon) self.remove({ coupon: coupon.code });
+
+    // TODO: Should this emit an 'unset' event?
+    // Need to consider other objects as well
+    if (!couponCode) {
+      debug('unset.coupon');
+      delete self.items.coupon;
+      return resolve();
```

Comment:
This adds a new class of events: `unset` -- I'd like to follow this PR with one adding `unset` events to the remaining objects.